### PR TITLE
[Themes] Add ability to change catalog theme

### PR DESCRIPTION
**Pull request by jgunaratne**
Exploration of how to potentially theme catalog.

simulator screen shot jun 2 2017 12 15 25 pm
simulator screen shot jun 2 2017 12 15 43 pm

### CODE REVIEW

**Review by ianegordon — CHANGES_REQUESTED** [review]
Works as expected locally, but Travis is throwing up some suspicious errors.  (Not the typical timeout)

I've kicked off Travis again to see how it goes.  If Travis is happy, LGTM.  Otherwise let me take another look tomorrow morning and figure out where things are falling down.

**Comment by ianegordon** [comment]
We need to address the default color issue with internal clients before we land this PR.

File: catalog/MDCCatalog.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -21,7 +21,7 @@
 		64307B431DF74FEF004AE4AC /* MDCActionExtension.appex in Embed App Extensions */ = {isa = PBXBuildFile; fileRef = 64307B381DF74FEF004AE4AC /* MDCActionExtension.appex */; settings = {ATTRIBUTES = (RemoveHeadersOnCopy, ); }; };
 		64307B4A1DF76198004AE4AC /* Assets.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 64307B491DF76198004AE4AC /* Assets.xcassets */; };
 		64BD963B1EBC9EF600284D1C /* MDCCatalogStyle.m in Sources */ = {isa = PBXBuildFile; fileRef = 64BD963A1EBC9EF600284D1C /* MDCCatalogStyle.m */; };
-		664524B71C6BA62A001ADBF8 /* AppDelegate.swift in Sources */ = {isa = PBXBuildFile; fileRef = 664524B61C6BA62A001ADBF8 /* AppDelegate.swift */; };
+		664524B71C6BA62A001ADBF8 /* MDCAppDelegate.swift in Sources */ = {isa = PBXBuildFile; fileRef = 664524B61C6BA62A001ADBF8 /* MDCAppDelegate.swift */; };
```
Comment (randallli) [inline catalog/MDCCatalog.xcodeproj/project.pbxproj]: revert rename of appdelegate

File: components/ActivityIndicator/examples/ActivityIndicatorExample.m [hunk 2]
```diff
@@ -17,7 +17,10 @@
 #import <UIKit/UIKit.h>
 
 #import "ActivityIndicatorExampleSupplemental.h"
+
+#import "MDCActivityIndicatorColorThemer.h"
```
Comment (randallli) [inline components/ActivityIndicator/examples/ActivityIndicatorExample.m]: lets not make the examples depend on the themer. (it doesn't look like this is needed anyways)

File: components/ActivityIndicator/examples/supplemental/ActivityIndicatorExampleSupplemental.h [hunk 3]
```diff
@@ -23,6 +23,7 @@
 
 #import "MaterialActivityIndicator.h"
 #import "MaterialCollections.h"
+#import "MaterialThemes.h"
```
Comment (randallli) [inline components/ActivityIndicator/examples/supplemental/ActivityIndicatorExampleSupplemental.h]: delete

File: components/AppBar/examples/AppBarTypicalUseExample.m [hunk 4]
```diff
@@ -17,6 +17,7 @@
 #import <UIKit/UIKit.h>
 
 #import "MaterialAppBar.h"
+#import "MaterialThemes.h"
```
Comment (randallli) [inline components/AppBar/examples/AppBarTypicalUseExample.m]: delete all the imports of materialThemes.

File: components/Buttons/examples/ButtonsTypicalUse.m [hunk 5]
```diff
@@ -16,7 +16,9 @@
 
 #import "ButtonsTypicalUseSupplemental.h"
 
+#import "MDCButtonColorThemer.h"
```
Comment (randallli) [inline components/Buttons/examples/ButtonsTypicalUse.m]: delete

**Review by randallli — DISMISSED** [review]

**Comment by ianegordon** [comment]
Unable to repro locally.
Please merge, but keep an eye out for smoke / fire.

**Review by ianegordon — APPROVED** [review]

**Pull request merged**